### PR TITLE
Next gen

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Changes to make twilio-chat work with next-gen libs.
⊂(ʘ‿ʘ)つ

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -16,10 +16,10 @@ app.set('views', path.join(__dirname, 'views'));
 app.set('view engine', 'jade');
 
 // uncomment after placing your favicon in /public
-//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
+// app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
```
Comment (hortega) [inline app.js]: Lines 18 and 19 can be deleted as most likely they won't be used and if they are needed, they can be recovered from history

File: routes/token.js [hunk 2]
```diff
@@ -2,7 +2,8 @@ var express = require('express');
 var router = express.Router();
 var TokenService = require('../services/tokenService');
 
-// POST /token/generate
+
+// POST /token
```
Comment (hortega) [inline routes/token.js]: :+1: for fixing the comment

File: test/token.js [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
-  , supertest = require('supertest')
-  , app = require('../app.js');
+var expect = require('chai').expect;
+var supertest = require('supertest');
+var app = require('../app.js');
 
-describe('token', function () {
-  describe('POST /token', function () {
-    it('generates a token', function (done) {
+
```
Comment (hortega) [inline test/token.js]: I think we could remove this extra line

File: package.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "private": "private",
   "scripts": {
     "start": "node ./bin/www",
-    "test": "export NODE_ENV=test && mocha test"
+    "test": "NODE_ENV=test npm test"
```
Comment (hortega) [inline package.json]: In this line you define what the command `npm test` do. If you write here `npm test` you are creating a loop and that's why your build is broken. Please replace back with `mocha`

File: package.json [hunk 5]
```diff
@@ -3,8 +3,8 @@
   "version": "1.0.0",
   "private": "private",
   "scripts": {
-    "start": "node ./bin/www",
-    "test": "NODE_ENV=test npm test"
+    "start": "npm start",
```
Comment (hortega) [inline package.json]: What's the reason for this change? It was fine before, but now you are referencing `npm start` from `npm start`, creating an endless loop

File: package.json (hunk 4 above)
Comment (acamino) [inline package.json]: Additinally, you should rely on the mocha version installed for this project, concretely in `./node_modules/mocha/bin/mocha test`.

There is another subtle problem. You are relying on _dotenv-safe_, which means that you don't have any step where you instruct the user to export the required env variables.

To fix this problem you have to explicitly export the env vars in the `test` script.

```javascript
"test": "NODE_ENV=test TWILIO_ACCOUNT_SID=ACXXXXXXX ...."
```

For referencing _mocha_ please use `./node_modules/.bin/mocha`.

**Comment by hortega** [comment]
thanks @juancarlospaco ! Looks good to me

File: package.json [hunk 6]
```diff
@@ -23,24 +23,26 @@
...
-    "kontainer-di": "^0.9.8",
-    "morgan": "~1.6.1",
-    "serve-favicon": "~2.3.0",
-    "twilio": "^3.3.0-edge"
+    "jquery": "^2.2.0",
+    "kontainer-di": "^0.9.9",
+    "moment": "^2.11.1",
+    "morgan": "~1.7.0",
+    "serve-favicon": "~2.3.2",
+    "twilio": "3.0.0-rc.16"
```
Comment (hortega) [inline package.json]: You could add `~` in front on the version so that it's not locked to this RC.
